homeassistant/components/tado/water_heater.py

```diff
@@ -0,0 +1,172 @@
+"""
+Support for Tado hot water zones.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/water_heater/tado/
+"""
+import logging
+
+from homeassistant.components.water_heater import (
+    SUPPORT_OPERATION_MODE,
+    WaterHeaterDevice,
+)
+from homeassistant.const import STATE_OFF, STATE_ON, TEMP_CELSIUS
+from homeassistant.components.tado import DATA_TADO
+
+_LOGGER = logging.getLogger(__name__)
+
+CONST_MODE_SMART_SCHEDULE = "SMART_SCHEDULE"
+CONST_MODE_OFF = "OFF"
+CONST_OVERLAY_TADO_MODE = "TADO_MODE"
+
+SUPPORT_FLAGS_HEATER = SUPPORT_OPERATION_MODE
```

Comment:
I've tried these changes, but the `water_heater.water_heater` entity that is created doesn't allow to change the temperature. I do have a `climate.hot_water` that does allow this. I guess this is because the `SUPPORT_TARGET_TEMPERATURE` flag is missing here.

Reply:
I correct myself, the interface allows this for the `climate.hot_water`, but the bridge replies with a 422 error.

Reply:
Ahh, that's interesting... My hot water zone does not do temperature control, so I don't have a climate zone for it and didn't add in temp control to this component.
Does the tado app let you control temperature for your hot water zone then? I may need some extra details in order to add support for that here.

Reply:
Indeed. I have temperature control, but can't see the current temperature. I've send you an email with more details.

Reply:
Added temperature control -- I think! Unable to test myself, so would appreciate if you can check the latest version out and provide feedback since yours has temp control.
Will update description to mention this is a breaking change for those with temp control on hot water zones.